### PR TITLE
Improvements

### LINKED ISSUE #1: Value in decimal is not considered
Regex should also match numbers after decimal point.
Screenshot from 2019-06-04 09-17-39
Screenshot from 2019-06-04 09-17-04

### LINKED ISSUE #3: Sites don't work when extension is installed
After installing the extension, sites like youtube.com are not opening properly. I think the reason behind this is that the extension is reading the content, and then trying to write it back.

### PR DESCRIPTION
**Pull request by ajeet97**
- Detect number after decimal places (Fixes #1)
- Detect formatted price (ex. `$ 1,000,000`)
- Format price after converting
- add new currency (EURO)
- Extension works when the extension icon is clicked
- Searches `document.body` using `BFS` and replaces desired elements only (fixes #3)

### CODE REVIEW

File: src/js/contentScript.js [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
-	console.log('content script loaded');
-	
-	const usdRegex = /\$( ?[0-9]+)/g;
-	const usdToRupeeRatio = 69.58;
+	const toRupeeRatio = {
+		'$': 69.58,
+		'€': 77.82,
+	};
+	const currencySymbols = ['\\$', '€'];
+	const currencyRegex = new RegExp(`(${currencySymbols.join('|')})\\s?(\\d+(?:,\\d{2,3})*(?:\\.\\d+)?)`, 'g');
```
Comment (rajatgoyal715) [inline src/js/contentScript.js:30]: Can we also group space, if any, so that `$1` becomes `₹69.58` not `₹ 69.58`?

File: src/js/contentScript.js [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
-		return Number.isNaN(numval) ? `$ ${val}` : `₹ ${usdToRupeeRatio * numval}`;
+	const INR = new Intl.NumberFormat('en-IN', {style: 'currency', currency: 'INR'});
+
+	const spanStyle = 'background: rgba(220, 220, 220, 0.7); color: #333; padding: 2px 5px; border-radius: 3px';
+	const format = price => `<span style="${spanStyle}">${INR.format(price)}</span>`;
+
+	document.body.innerHTML = document.body.innerHTML.replace(currencyRegex, (_, currency, val) => {
+		const numval = Number(val.replace(/,/g, ''));
+		return Number.isNaN(numval) ?
+			`${currency} ${val}` :
```
Comment (rajatgoyal715) [inline src/js/contentScript.js]: And we can use that space here like `${currency}${space}${val}`

**Review by rajatgoyal715 — CHANGES_REQUESTED** [review]
Everything looks good. I have suggested a little improvement.

File: src/js/contentScript.js (hunk 1 above)
Comment (ajeet97) [reply src/js/contentScript.js:30]: @rajatgoyal715 The prices are formatted using standard conventions. So I think it is good to stick to the standard conventions.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# 0.0.2
+ - Detect number after decimal places
+ - Detect formatted price (ex. $ 1,000,000)
+ - Format price after converting
+ - add new currency (EURO)
```
Comment (rajatgoyal715) [inline CHANGELOG.md:5]: Can make the first character a capital one :p

File: src/js/contentScript.js [hunk 4]
```diff
@@ -1,11 +1,45 @@
...
-	
-	const usdRegex = /\$( ?[0-9]+)/g;
-	const usdToRupeeRatio = 69.58;
-
-	document.body.innerHTML = document.body.innerHTML.replace(usdRegex, (_, val) => {
-		const numval = Number(val);
-		return Number.isNaN(numval) ? `$ ${val}` : `₹ ${usdToRupeeRatio * numval}`;
+	function replace($node, regex, replaceMethod) {
+		const queue = [$node];
+		while (queue.length) {
```
Comment (rajatgoyal715) [inline src/js/contentScript.js:4]: Are we doing this, so that headlines of search results get converted first, and then the descriptions of all results?

**Review by rajatgoyal715 — APPROVED** [review]

**Comment by rajatgoyal715** [comment]
@ajeet97 Are there any blockers on this? Or can we merge this?

**Comment by ajeet-smpx** [comment]
yes we can merge this.

**Pull request merged**